### DISCUSSION TITLE
Provide support for user preferences

### DISCUSSION THREAD

**Discussion by mgifford** [uswds/uswds#5798] [Accessibility]
We have to move beyond the idea that one design will be the "most accessible". People can have contradictory needs, and our technology can accommodate that. All major operating systems and browsers support a range of personalization features which are not supported by the USWDS. 

**prefers-color-scheme**

- https://caniuse.com/prefers-color-scheme
- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
- https://www.smashingmagazine.com/2023/08/css-accessibility-inclusion-user-choice/
- https://css-tricks.com/almanac/properties/c/color-scheme/

**prefers-contrast**

- https://caniuse.com/mdn-css_at-rules_media_prefers-contrast
- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast
- https://www.smashingmagazine.com/2023/01/creating-high-contrast-design-system-css-custom-properties/
- https://css-tricks.com/a-complete-guide-to-css-media-queries/

**prefers-reduced-motion**

- https://caniuse.com/prefers-reduced-motion
- https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion
- https://www.smashingmagazine.com/2021/10/respecting-users-motion-preferences/
- https://web.dev/articles/prefers-reduced-motion

I think we should start with Dark Mode. Not everyone will benefit from it, but respecting users choice is always a good idea. Many users have specified that they prefer Dark Mode, so why ignore that? 

- https://css-tricks.com/a-complete-guide-to-dark-mode-on-the-web/
- https://blogs.windows.com/msedgedev/2021/06/16/dark-mode-html-form-controls/
- https://medium.com/eightshapes-llc/light-dark-9f8ea42c9081
- https://blog.zenhub.com/improving-design-system-for-dark-mode-theming/
- https://uxdesign.cc/dark-ui-design-principles-and-best-practices-9b9061b86e1https://uxdesign.cc/designing-dark-mode-c7bd200eaa64

So how do we build the USWDS to provide better support for user preferences.

**Comment by caseywatts** [comment]
My team is talking about dark mode a bit this week, +1 to this idea!

I also had a user interview a month ago with a low-vision user, and they said that bright websites are significantly harder for them to read. They asked us to support dark mode, too.

If USWDS were to support dark mode I think my team is one that would adopt it!

**Comment by katannshaw** [comment]
I 100% agree. I think a move to a 1:1 match for USWDS Sass variables to CSS variables would resolve this issue. It looks like it was discussed [during a USWDS monthly call on July 11, 2024](https://github.com/uswds/uswds/discussions/5977#discussioncomment-10023845). I wonder what the progress is.

**Reply by caseywatts** [reply]
There has been some progress on this lately -- css variables would definitely help a ton with being able to support dark mode!
https://github.com/uswds/uswds-elements/issues/141
